pkg/controller/deployment/default_cert_manager_creator.go

```diff
@@ -0,0 +1,65 @@
+package deployment
+
+import (
+	"context"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/cert-manager-operator/apis/operator/v1alpha1"
+	alpha1 "github.com/openshift/cert-manager-operator/pkg/operator/clientset/versioned/typed/operator/v1alpha1"
+)
+
+type DefaultCertManagerController struct {
+	operatorClient    v1helpers.OperatorClient
+	controllerFactory *factory.Factory
+	recorder          events.Recorder
+	name              string
+	certManagerClient alpha1.OperatorV1alpha1Interface
+}
+
+func (c *DefaultCertManagerController) Name() string {
```

Comment:
curious: why is this needed?

Reply:
Required by the `factory.Controller` interface.

Reply:
this is returning an instance created by the controller factory, `Name()` is implemented by `baseController`. The `Name()` function here has no effect.

Reply:
https://github.com/openshift/cert-manager-operator/blob/b9570ecc15835f9e3b266957f0e0be01b97e1a09/vendor/github.com/openshift/library-go/pkg/controller/factory/base_controller.go#L45-L47

Reply:
Removed